### ISSUE TITLE
Variation in FHIRpath identification of extensions

### ISSUE THREAD

**Issue by robeastwood-agency** [hl7au/au-fhir-base#250]
### Overview
We have been doing alot of work with invariants FHIRpath in Agency profiles of late and noticed that we were identifying extensions in FHIRpath in different ways. As a result of digging into the spec, we have consequently decided to use a uniform approach. We have also noticed that the same variation exists in the HL7 AU base IG as well. Raising an issue more for awareness, rather than as anything wrong, as the variations reach the same outcome (we believe). The only consequence of the variation is that informed readers may be confused by the different approach.

### The variations
```
Composition.extension.where(url='http://hl7.org.au/fhir/StructureDefinition/composition-author-role').exists()
vs
Composition.extension('http://hl7.org.au/fhir/StructureDefinition/composition-author-role').exists()
```
We have concluded from our testing that both have the same result and according to the [FHIRpath spec](http://hl7.org/fhirpath/#additional-functions), they are equivalent but one is a shortcut for the other:
> **B.3.1. extension(url : string) : collection**
> Will filter the focus for items named "extension" with the given url. This is a syntactical shortcut for `.extension.where(url = string)`, but is simpler to write. Will return an empty collection if the focus is empty or the url is empty.

#### au-address
[inv-add-1](http://build.fhir.org/ig/hl7au/au-fhir-base/StructureDefinition-au-address-definitions.html#Address) 
`address.extension.where(url='http://hl7.org.au/fhir/StructureDefinition/no-fixed-address' and value=true).exists() implies address.type='physical'`

[inv-add-2](http://build.fhir.org/ig/hl7au/au-fhir-base/StructureDefinition-au-address-definitions.html#Address) 
`address.extension.where(url='http://hl7.org.au/fhir/StructureDefinition/no-fixed-address').exists() implies address.text.startsWith('NO FIXED ADDRESS')`

#### au-patient (mixed)
[inv-pat-0](http://build.fhir.org/ig/hl7au/au-fhir-base/StructureDefinition-au-patient-definitions.html#Patient) 
`birthDate.extension.where(url='http://hl7.org/fhir/StructureDefinition/patient-birthTime').exists() implies birthDate.extension('http://hl7.org/fhir/StructureDefinition/patient-birthTime').value.toString().substring(0,10) = birthDate.toString()`

#### au-allergyintolerance
[inv-ait-0](http://build.fhir.org/ig/hl7au/au-fhir-base/StructureDefinition-au-allergyintolerance-definitions.html#AllergyIntolerance) 
`(extension('http://hl7.org.au/fhir/StructureDefinition/recorder-related-person').exists() and recorder.exists()).not()`

#### au-composition
[inv-com-0](http://build.fhir.org/ig/hl7au/au-fhir-base/StructureDefinition-au-composition-definitions.html#Composition.attester)
`(attester.extension('http://hl7.org.au/fhir/StructureDefinition/attester-related-party').exists() and attester.party.exists()).not()`

#### au-medlist
[inv-lst-0](http://build.fhir.org/ig/hl7au/au-fhir-base/StructureDefinition-au-medlist-definitions.html#List) 
`(extension('http://hl7.org.au/fhir/StructureDefinition/list-source-role').exists() and source()).not()`

### Proposal
Consider moving to the simpler approach to identify extensions, to facilitate readability.

(Note - as we had a hand in most of the above profiles, I suspect that we introduced the inconsistencies)

**Comment by oridashi** [comment]
review

**Comment by robeastwood-agency** [comment]
AU Base Patient profile has been updated (via https://github.com/hl7au/au-fhir-base/commit/20a035882f1ee9ff7a95210faa12251a2c4781b1) with the simplified version of the FHIRPath.

Leaving only the 2 AU Address profile invariants. These will be updated in a forthcoming PR (low priority)

**Comment by robeastwood-agency** [comment]
This can now be closed following the merge of PR #431
